demoapp/src/main/java/me/uport/sdk/demoapp/request_flows/TypedDataRequestActivity.kt

```diff
@@ -0,0 +1,131 @@
+package me.uport.sdk.demoapp.request_flows
+
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.view.View
+import com.uport.sdk.signer.KPSigner
+import kotlinx.android.synthetic.main.request_flow.*
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import me.uport.sdk.core.UI
+import me.uport.sdk.demoapp.R
+import me.uport.sdk.jwt.JWTTools
+import me.uport.sdk.transport.Transports
+import org.json.JSONObject
+
+/**
+ *
+ * This activity demonstrates the flow for creating and sending a [Typed Data Request]
+ *
+ **/
+
+class TypedDataRequestActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.request_flow)
+
+        // instantiate signer
+        val signer = KPSigner("0x1234")
+
+        // create issuer DID
+        val issuerDID = "did:ethr:${signer.getAddress()}"
+
+        // create the request JWT payload
+        @Suppress("StringLiteralDuplication")
+        val payload = mapOf<String, Any>(
+                "callback" to "https://uport-project.github.io/uport-android-sdk",
+                "type" to "eip712Req",
+                "net" to "0x4",
+                "iss" to issuerDID,
+                "iat" to System.currentTimeMillis(),
+                "typedData" to mapOf(
```

Comment:
Good call showing how to build these requests without the actual support from SDK static types.